docs/scaling.md

```diff
@@ -0,0 +1,265 @@
+
+```python
+from concurrent.futures import Executor
+
+class CustomExecutor(Executor):
+    def map(
+        self,
+        fn: Callable,
+        *iterables: Iterable,
+    ) -> Iterator:
+        ...
+
+combined_vds = vz.open_virtual_mfdataset(filepaths, parallel=CustomExecutor)
+```
+
+## Memory usage
+
+For the virtualization to succeed you need to ensure that your available memory is not exceeded at any point.
+There are 3 points at which this might happen:
+
+1. While generating references
+2. While combining references
+3. While writing references
+
+While generating references each worker calling `open_virtual_dataset` needs to avoid running out of memory.
+This primarily depends on how the file is read - see the section on [Caching](scaling.md#Caching) below.
+
+The combine step happens back on the machine on which `vz.open_virtual_mfdataset` was called, so while combining references that machine must have enough memory to hold all the virtual references at once.
+You can find the in-memory size of the references for a single virtual dataset by calling the `.nbytes` accessor method on it (not to be confused with the `.nbytes` xarray method, which returns the total size if all that data were actually loaded into memory).
+Do this for one file, and multiply by the number of files you have to estimate the total memory required for this step.
+
+Writing the combined virtual references out requires converting them to a different references format, which may have different memory requirements.
+
+## Scalability of references formats
+
+After the map-reduce operation is complete, you will likely still want to persist the virtual references in some format.
+Depending on the format, this step may also have scalability concerns.
+
+### Kerchunk
+
+The Kerchunk references specification supports 3 formats - an in-memory (nested) `dict`, JSON, and Parquet.
+
+Both the in-memory Kerchunk `dict` and Kerchunk JSON formats are extremely inefficient ways to represent virtual references.
+You may well find that a virtual dataset object that easily fits in memory suddenly uses up many times more memory or space on disk when converted to one of these formats.
+Persisting large numbers of references in these formats is therefore not recommended.
+
+The Kerchunk Parquet format is more scalable, but you may want to experiment with the  `record_size` and `categorical_threshold` arguments to the virtualizarr `.to_kerchunk` accessor method.
+
+### Icechunk
+
+[Icechunk](https://icechunk.io/) uses it's own [open format](https://icechunk.io/en/latest/spec/) for persisting virtual references.
+
+Icechunk's format stores the virtual references in dedicated binary files, and can use "manifest splitting", together meaning that it should be a scalable way to store large numbers of references.
+
+TODO: Put numbers on this by testing at large scale once manifest splitting is actually released in Icechunk.
```

Comment:
```suggestion
!!! TODO
    Put numbers on this by testing at large scale once manifest splitting is actually released in Icechunk.
```
Can we put the TODOs in admonitions so they can be more easily skipped over when reading the pages?